src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = false || true;
```

Comment:
I think you tried to make it a toggle. If so you can write `this.hidden = !this.hidden` otherwise `this.hidden = true` is enough.

Reply:
That's good. It had to be a toggle, yes

Reply:
@lefteris-L what does this statement ALWAYS return?
`false || true` - it will alway return true. So it does not make sense.
You need to fix such things.
```suggestion
    this.hidden = true;
```